src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -0,0 +1,293 @@
+
+bool PerformanceConfigHipImplicitGemmFwdXdlops::IsValid(const ConvolutionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    std::vector<DeviceConvFwdPtr_t> conv_ptrs;
+    add_device_conv2d_fwd_xdl_nhwc_kyxc_nhwk_int8_instances_t(conv_ptrs);
+    const auto args   = CKArgs{ctx};
+    auto argument_ptr = conv_ptrs[this->index].MakeArgumentPointer(nullptr,
+                                                                   nullptr,
+                                                                   nullptr,
+                                                                   args.N,
+                                                                   args.K,
+                                                                   args.C,
+                                                                   args.input,
+                                                                   args.filter,
+                                                                   args.input,
+                                                                   args.strides,
+                                                                   args.dilation,
+                                                                   args.lPadding,
+                                                                   args.rPadding);
+    return conv_ptrs[this->index].IsSupportedArgument(argument_ptr.get());
+#endif
+}
+
+bool PerformanceConfigHipImplicitGemmFwdXdlops::operator==(
+    const PerformanceConfigHipImplicitGemmFwdXdlops& other) const
+{
+    return this->index == other.index;
+}
+
+PerformanceConfigHipImplicitGemmFwdXdlops
+ConvHipImplicitGemmFwdXdlops::GetDefaultPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    PerformanceConfigHipImplicitGemmFwdXdlops pp;
+    pp.HeuristicInit(ctx);
+    return pp;
+}
+
+bool ConvHipImplicitGemmFwdXdlops::IsValidPerformanceConfig(
+    const ConvolutionContext& ctx, const PerformanceConfigHipImplicitGemmFwdXdlops& config) const
+{
+    return config.IsValid(ctx);
+}
+
+PerformanceConfigHipImplicitGemmFwdXdlops
+ConvHipImplicitGemmFwdXdlops::Search(const ConvolutionContext& ctx,
+                                     const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, invoke_ctx);
+}
+
+size_t ConvHipImplicitGemmFwdXdlops::GetWorkspaceSize(const ConvolutionContext& ctx) const
+{
+    std::ignore = ctx;
+    return 0;
+}
+
+bool ConvHipImplicitGemmFwdXdlops::IsApplicable(const ConvolutionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(!ctx.use_hip_kernels)
+        return false;
```

Comment:
```suggestion
```
MIOPEN_USE_HIP_KERNELS and all the underlying machinery (down to `use_hip_kernels`) is intended for disabling kernels that require _online HIP compilation_. AFAICS CK contains precompiled kernels. So I think we do not need this.

Reply:
Thanks for catching this.

Reply:
Resolved!